Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp

```diff
@@ -211,18 +211,19 @@ void ParticleHistogram::ComputeDiags (int step)
                 {
                     amrex::ParticleReal x, y, z;
                     GetPosition(i, x, y, z);
-                    auto const w  = d_w[i];
+                    auto const w  = (amrex::Real)d_w[i];
```

Comment:
For all the things below, I would rather stay/cast to `ParticleReal`. But is it possible that all the functor stuff accepts only `Real`?

Reply:
Are you concerned about execution cost, or how the code is written? This could be
```suggestion
                    amrex::Real const w  = d_w[i];
```
If it's about the expression

IIRC, the parsers are generic and could be passed potentially single-precision `ParticleReal`, but there's a comment somewhere saying we should stick with double-precision for calls to parsed functions, maybe by Remi.

Reply:
The `fun_filterparser` and `fun_partparser` stuff below can stay as `ParticleReal`, I'll change it back that way.

Reply:
@PhilMiller I am concerned about the cost of a cast on GPU, it's a special instruction and "surprisingly" costly.

Reply:
Today I learned. Thanks

Reply:
Should I leave this line as is or remove the explicit cast and change the line to be what Phil suggested `amrex::Real const w = d_w[i];`?

Reply:
You should probably leave it as is, to emphasize that a cast operation is occurring, which may be a noticeable expense.